src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileInterface/ResourceOptimizationOpenshiftImpl.java

```diff
@@ -83,36 +80,26 @@ public List<KruizeRecommendationEngine> getEngines() {
     @Override
     public void generateRecommendation(KruizeObject kruizeObject, List<ExperimentResultData> experimentResultDataList, Timestamp interval_start_time, Timestamp interval_end_time) throws Exception {
         /*
-            To restrict the number of rows in the result set, the Load results operation involves locating the appropriate method and configuring the desired limitation.
-            It's important to note that in order for the Limit rows feature to function correctly,
-            the CreateExperiment API must adhere strictly to the trail settings' measurement duration and should not allow arbitrary values
+             The general strategy involves initially attempting the optimal query;
         */
-        String experiment_name = kruizeObject.getExperimentName();
-        int theoriticalLimitRows = (int) ((
-                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS *
-                        KruizeConstants.DateFormats.MINUTES_FOR_DAY)
-                / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
-
-        // Adding a 10% of value to make sure there are enough data points to gather the duration required
-        // Will be deprecated once we extract exact number of rows from DB based on duration in minutes needed for each term
-        int practicalLimitRows = RecommendationUtils.getThreshold(theoriticalLimitRows, 10, true);
-
-        if (null != interval_start_time) {
-            long diffMilliseconds = interval_end_time.getTime() - interval_start_time.getTime();
-            long minutes = diffMilliseconds / (60 * 1000);
-            int addToLimitRows = (int) (minutes / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
-            LOGGER.debug("add to limit rows set to {}", addToLimitRows);
-            practicalLimitRows = practicalLimitRows + addToLimitRows;
-        }
-        LOGGER.debug("Limit rows set to {}", practicalLimitRows);
-
-
+        // Convert the Timestamp to a Calendar instance in UTC time zone
+        Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        cal.setTimeInMillis(interval_end_time.getTime());
+        /*
+         * interval_start_time Subtract (LONG_TERM_DURATION_DAYS +  THRESHOLD days)
+         * Incorporate a buffer period of "threshold days" to account for potential remote cluster downtime.
+         * This adjustment aims to align the cumulative hours' duration with LONG_TERM_DURATION_DAYS.
+         */
+        cal.add(Calendar.DAY_OF_MONTH, -(KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS +
+                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS_THRESHOLD));
```

Comment:
I think rather than adding a threshold and hoping it will work for all cases, I think this needs to be converted to a loop. After each iteration we can check if we have the required number of hours, if not we try again with the same quantum again by going further back in time. This can stop after we have reached a limit or there are no more rows, in which case, we return the "NOT_ENOUGH_DATA" error